elastic/kibana
# [RAM] Slack connector improvements

Issue: https://github.com/elastic/kibana/issues/150195

## Summary

In this PR I have made changes to Slack connector.
Previously we could used Slack Webhook to send messages to a sertain Slack channel.
One Connector - one webhook - one Slack channel.

Customers asked for opportunity to point one Slack connector to different Slack channels.
So here in this PR I introduced Slack Web API to get channels list and post messages to channels. 
At the same time I keep old webhook solution in place as well to have backward compatibility.
As the result now we have 2 options during Slack connector creation: Web API and Webhook:
<img width="950" alt="Screenshot 2023-02-16 at 12 14 35" src="https://user-images.githubusercontent.com/26089545/219367378-0d7a7525-9f88-4143-a34b-fed6ff4756be.png">

For using Web API option, you need to have Slack API token. We do not have official docs, but I've written some instructions here for now: https://docs.google.com/document/d/1vEuKg7Yb7tejRYzQMxY_oQIeCKxF1cOW2QK61GtqhNM/edit?usp=sharing

How to test:
1) Create Slack App using google doc above (create new workspace for that purpose) and get API token.
2) Try to create connector from Management -> Connectors with both Webhook and Web API options. Try to send messages
![Screenshot 2023-02-16 at 15 10 09](https://user-images.githubusercontent.com/26089545/219387745-0863dd82-3cdc-4d62-9b64-3980cbaba294.png)
3) Create rules and create connector from this form as well. 
<img width="950" alt="Screenshot 2023-02-16 at 12 14 35" src="https://user-images.githubusercontent.com/26089545/219387459-32dfec13-f030-4330-b3e1-09e44958d0cf.png">
![Screenshot 2023-02-16 at 15 17 57](https://user-images.githubusercontent.com/26089545/219389603-4d2e5a15-31e4-4f01-8232-bb1e6f145dbc.png)
4) Check if alert will be trigger, you'll have messages in the Slack

Sometimes Slack API call needs to be tried. It need more work, which will be done as followup. So now it's possible to pick only one channel in UI. It'll change in the future.

### Checklist
